cluster-provider-client/provider.go

```diff
@@ -29,99 +30,83 @@ import (
-							if err != nil {
-								logger.Error(err, "")
-								// TODO: how do we handle failure?
-								return
-							}
-						}
-					case watch.Deleted:
-						// TODO: I am currently ignoring the possibility of the logical cluster object not existing
-						var nameLogicalClusterObject string = ""
-						for _, logicalCluster := range listLogicalClusters.Items {
-							if compareLogicalClusters(logicalCluster, eventLogicalCluster) {
-								nameLogicalClusterObject = logicalCluster.Name
-								break
-							}
-						}
-						if nameLogicalClusterObject != "" {
-							logger.Info("Deleting LogicalCluster object", nameLogicalClusterObject)
-							err := clientset.LogicalclusterV1alpha1().LogicalClusters(providerName).Delete(ctx, nameLogicalClusterObject, v1.DeleteOptions{})
-							if err != nil {
-								logger.Error(err, "")
-								// TODO: how do we handle failure?
-								return
-							}
-						}
-
-					default:
-						// Unknown!
-						logger.Info("unknown event")
-						// TODO return an error or panic?
-					}
+	for {
+		event, ok := <-w.ResultChan()
+		if !ok {
+			w.Stop()
+			logger.Info("stopping")
+			// TODO: return an error
+			return
+		}
+		listLogicalClusters, err := clientset.LogicalclusterV1alpha1().LogicalClusters(providerName).List(ctx, v1.ListOptions{})
+		if err != nil {
+			logger.Error(err, "")
+			// TODO: how do we handle failure?
+			return
+		}
+		switch event.Type {
+		case watch.Added:
+			// TODO: I am currently ignoring the possibility of the logical cluster object already existing
+			var found bool = false
+			for _, logicalCluster := range listLogicalClusters.Items {
+				if logicalCluster.Name == event.Name {
+					found = true
+					break
+				}
+			}
+			if !found {
+				logger.Info("Creating new LogicalCluster object", event.Name)
+				var eventLogicalCluster v1alpha1apis.LogicalCluster
+				eventLogicalCluster.Name = event.Name
+				eventLogicalCluster.Spec.ClusterProviderDesc = providerName
+				eventLogicalCluster.Spec.Managed = false
+				eventLogicalCluster.Status.Phase = "Initializing"
+				_, err = clientset.LogicalclusterV1alpha1().LogicalClusters(providerName).Create(ctx, &eventLogicalCluster, v1.CreateOptions{})
+				if err != nil {
+					logger.Error(err, "")
+					// TODO: how do we handle failure?
+					return
 				}
-			}()
+			}
+		case watch.Deleted:
+			logger.Info("Deleting LogicalCluster object", event.Name)
+			err := clientset.LogicalclusterV1alpha1().LogicalClusters(providerName).Delete(ctx, event.Name, v1.DeleteOptions{})
+			if err != nil {
+				// TODO: If the logical cluster object does not exist, ignore the error.
+				logger.Error(err, "")
+				// TODO: how do we handle failure?
+				return
+			}
+
 		default:
-			panic("unknown provider type")
+			// Unknown!
+			logger.Info("unknown event")
+			// TODO return an error or panic?
 		}
 	}
+}
 
+// ES: return and error and don't panic, move push to outside the case
```

Comment:
please add a comment that this is temporary for Stage 1. For stage 2 we have a uniform interface for all providers